### PR TITLE
Initial shop logic

### PR DESCRIPTION
**Pull request by Ido-Barnea**
Well, that was simple. All that's left is to create the shop's UI (#113).

### CODE REVIEW

File: development/code/LogicAdapter.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ export function onActionTriggered(
     onPlayerAction(draggedPiece, targetPiece);
   } else if (targetElement.classList.contains('item')) {
     game.getItems().forEach((item) => {
-      if (comparePositions(item.position, targetElementPosition)) {
+      if (item.position && comparePositions(item.position, targetElementPosition)) {
```
[the remark this answers is not in the document]
Comment (Ido-Barnea) [reply development/code/LogicAdapter.ts:74]: I disagree, `comparePositions()` should have one job - to compare positions, not to check their validity.

[the remark this answers is not in the document]
Comment (Monkey-bored) [reply development/code/LogicAdapter.ts:74]: I think its a bit goofy that I had to go through every instance of `comparePositions()` in order to check if you forgot to put item.position.

[the remark this answers is not in the document]
Comment (Ido-Barnea) [reply development/code/LogicAdapter.ts:74]: It will throw an error if you don't add `item.position &&` but I see what you mean

**Review by Monkey-bored — APPROVED** [review]
LGTM 💵💶💷🤑💸

**Pull request merged**